stage/vpc/waf.tf

```diff
@@ -8,27 +8,138 @@ resource "aws_wafv2_web_acl" "stage-default" {
   }
 
   rule {
-    name     = "stageRateLimitingRule"
+    name     = "stageRateLimitAuthenticated"
     priority = 1
 
     action {
       block {}
     }
 
+    statement {
+      rate_based_statement {
+        limit              = 3000
+        aggregate_key_type = "IP"
+
+        scope_down_statement {
+          or_statement {
+            statement {
+              byte_match_statement {
+                search_string = "bearer "
```

Comment:
The problem we have with just checking for headers is that this is basically same as email rate limit. it's not really authenticated.
This is an okay solution I guess given we don't have anything else more sophisticated. Though now I'm thinking about it I dunno if we should do it at all... 
The only way to do this is going to be to get something to set a header to the incoming request after it's been checked for a valid auth.

Anyway for your PR fine given the limitations.